### PR TITLE
LANG-1406 StringIndexOutOfBoundsException in StringUtils.replaceIgnor…

### PR DESCRIPTION
**Pull request by drajakumar**
…eCase

### CODE REVIEW

**Comment by drajakumar** [comment]
@kinow can you kindly review and comment whether this fix holds good, thank you!

**Comment by PascalSchumacher** [comment]
@kinow What do you think is this a better solution than #340?

File: src/main/java/org/apache/commons/lang3/StringUtils.java [hunk 1]
```diff
@@ -5596,7 +5596,16 @@ private static String replace(final String text, String searchString, final Stri
...
+             for (int i = 0; i < text.length(); i++) {
+                 char c = text.charAt(i);
+                 if (Character.UnicodeBlock.of(c) != Character.UnicodeBlock.BASIC_LATIN) {
+                     buf.append(c);
+                 } else {
+                     buf.append(Character.toLowerCase(c));
+                 }
+             }
+             searchText = buf.toString();
              searchString = searchString.toLowerCase();
```
Comment (geratorres) [inline src/main/java/org/apache/commons/lang3/StringUtils.java]: searchString could contain one of those characters like "\u0130"

**Comment by garydgregory** [comment]
Needs conflicts resolved...

**Comment by kinow** [comment]
Oh, actually the issue has been closed in #423! Closing the PR, but thanks for your contribution @drajakumar !

**Pull request closed without merging**